### PR TITLE
feat: Send user groups to proxy extensions

### PR DESCRIPTION
**Pull request by leoluz**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: server/extension/extension.go [hunk 1]
```diff
@@ -709,7 +720,8 @@ func (m *Manager) CallExtension() func(http.ResponseWriter, *http.Request) {
 			return
 		}
 
-		prepareRequest(r, extName, app, m.username(r.Context()))
+		groups := m.groupsClaim(r.Context(), []string{"groups"})
```
Comment (agaudreault) [inline server/extension/extension.go:723]: The list of scopes is a configurable settings in ArgoCD rbac configMap and should be retrieved from the settings object.

**Review by agaudreault — CHANGES_REQUESTED** [review]

File: server/extension/extension.go (hunk 1 above)
Comment (leoluz) [reply server/extension/extension.go]: done

**Review by agaudreault — APPROVED** [review]
LGTM

File: .mockery.yaml [hunk 2]
```diff
@@ -43,6 +43,7 @@ packages:
       ProjectGetter:
       RbacEnforcer:
       SettingsGetter:
+      UserGetter:
```
Comment (agaudreault) [inline .mockery.yaml:46]: codegen might be missing for UserGetter ? 🤔

Comment (leoluz) [reply .mockery.yaml:46]: This is just the way mockery expects to have the list interfaces provided. If there is no additional configs for that mock, we should just provide the name of the interface followed by a colon.

**Review by wanghong230 — APPROVED** [review]

**Pull request merged**